test/arcade/core/util/SolverTest.java

```diff
@@ -0,0 +1,226 @@
+package arcade.core.util;
+
+import org.junit.jupiter.api.Test;
+import arcade.core.util.Solver.Equations;
+import arcade.core.util.Solver.Function;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SolverTest {
+    @Test
+    public void testConstructor_called_returnsException() {
+        assertThrows(
+                UnsupportedOperationException.class,
+                () -> {
+                    new Solver();
+                });
+    }
+
+    @Test
+    public void testEuler_calledWithEquations_returnsAnswer() {
+        Equations e =
+                new Equations() {
+                    public double[] dydt(double t, double[] y) {
+                        double[] result = new double[2];
+                        result[0] = 1;
+                        result[1] = -1;
+                        return result;
+                    }
+                };
+
+        double[] y = Solver.euler(e, 0, new double[] {0, 1}, 1, 0.1);
+        assertEquals(1, y[0], 0.01);
+        assertEquals(0.0, y[1], 0.01);
+    }
+
+    @Test
+    public void testEuler_calledWithComplexEquations_returnsAnswer() {
+        Equations e =
+                new Equations() {
+                    public double[] dydt(double t, double[] y) {
+                        double[] result = new double[2];
+                        result[0] = -y[0];
+                        result[1] = y[0];
+                        return result;
+                    }
+                };
+
+        double[] y = Solver.euler(e, 0, new double[] {1, 0}, 1, 0.01);
+        assertEquals(Math.exp(-1), y[0], 0.01);
+        assertEquals(1 - Math.exp(-1), y[1], 0.01);
+    }
+
+    @Test
+    public void testRungeKutta_calledWithEquations_returnsAnswer() {
```

Comment:
```suggestion
    public void testRungeKutta_calledWithSimpleEquations_returnsAnswer() {
```

For consistency with the "complex" tests?